[GitHub Actions] Fail fast from the default shells (bash/sh) in Linux/macOS (#27583)
github/docs: content/actions/using-workflows/workflow-syntax-for-github-actions.md

For built-in shell keywords, we provide the following defaults that are executed by {% data variables.product.prodname_dotcom %}-hosted runners. You should use these guidelines when running shell scripts.

- `bash`/`sh`:

>   - By default, fail-fast behavior is enforced using `set -e` for `sh` and `bash` both. When `shell: bash` is specified, `-o pipefail` is also applied to enforce early exit from pipelines in case of non-zero exit status.

Reviewer: ```suggestion
  - By default, fail-fast behavior is enforced using `set -e` for both `sh` and `bash`. When `shell: bash` is specified, `-o pipefail` is also applied to enforce early exit from pipelines that generate a non-zero exit status.
```
